pkg/agent/apiserver/handlers/bgproute/handler.go

```diff
@@ -68,10 +69,33 @@ func HandleFunc(bq querier.AgentBGPPolicyInfoQuerier) http.HandlerFunc {
 		}
 
 		var bgpRoutesResp []apis.BGPRouteResponse
-		for _, bgpRoute := range bgpRoutes {
-			bgpRoutesResp = append(bgpRoutesResp, apis.BGPRouteResponse{
-				Route: bgpRoute,
-			})
+		if !ipv4Only && !ipv6Only {
+			for bgpRoute := range bgpRoutes {
+				if net.IsIPv4CIDRString(bgpRoute.Prefix) {
+					bgpRoutesResp = append(bgpRoutesResp, apis.BGPRouteResponse{
+						Route:     bgpRoute.Prefix,
+						Type:      string(bgpRoutes[bgpRoute].Type),
+						K8sObjRef: bgpRoutes[bgpRoute].K8sObjRef,
+					})
+				}
+			}
+			for bgpRoute := range bgpRoutes {
+				if net.IsIPv6CIDRString(bgpRoute.Prefix) {
+					bgpRoutesResp = append(bgpRoutesResp, apis.BGPRouteResponse{
+						Route:     bgpRoute.Prefix,
+						Type:      string(bgpRoutes[bgpRoute].Type),
+						K8sObjRef: bgpRoutes[bgpRoute].K8sObjRef,
+					})
+				}
+			}
```

Comment:
I don't understand this code. We already pass `ipv4Only` and `ipv6Only` to `GetBGPRoutes`, so why is there logic here that depends on the IP family of the route prefix?
Also, I cannot see the difference between the 2 for loops.

Reply:
Hi @antoninbas 
It is to maintain the order when both types(ipv4 and ipv6) of routes are present, first all ipv4 routes, then all ipv6 routes in the output. Previously, slice of string was returned which already had this order maintained.

Reply:
Got it. Then let's keep the code more explicit and more comprehensive:
```go
for bgpRoute := range bgpRoutes {
        bgpRoutesResp = append(bgpRoutesResp, apis.BGPRouteResponse{
                Route:     bgpRoute.Prefix,
                Type:      string(bgpRoutes[bgpRoute].Type),
                K8sObjRef: bgpRoutes[bgpRoute].K8sObjRef,
        })
}
// make sure that we provide a stable order for the API response
slices.SortFunc(bgpRoutesResp, func(a, b apis.BGPRouteResponse) int {
        pA, _ := netip.ParsePrefix(a.Route)
        pB, _ := netip.ParsePrefix(b.Route)
        // IPv4 routes first, then IPv6 routes
        if pA.Addr().Is4() && !pB.Addr().Is4() {
                return -1
        }
        if !pA.Addr().Is4() && pB.Addr().Is4() {
                return 1
        }
        // both routes are from the same IP family, now order based on route type
        if n := strings.Compare(a.Type, b.Type); n != 0 {
                return n     
        }
        // finally, for routes of the same IP family and type, order based on prefix
        // shorter prefixes come first; if the length is the same we order by IP
        if n := pA.Bits() - pB.Bits(); n != 0 {
                return n
        }
        return pA.Addr().Compare(pB.Addr())
})
```

This way we will have a stable and meaningful order for the API response

Reply:
done, thanks @antoninbas